compiler/natives/src/reflect/reflect.go

```diff
@@ -1325,6 +1325,26 @@ func getJsTag(tag string) string {
 	return ""
 }
 
+func (v Value) grow(n int) {
+	if n < 0 {
+		panic(`reflect.Value.Grow: negative len`)
+	}
+
+	s := v.object()
+	len := s.Get(`$length`).Int()
+	if len+n < 0 {
+		panic(`reflect.Value.Grow: slice overflow`)
+	}
+
+	cap := s.Get(`$capacity`).Int()
+	if len+n > cap {
+		ns := js.Global.Call("$growSlice", s, len+n)
+		s.Set(`$capacity`, ns.Get(`$capacity`))
+		s.Set(`$array`, ns.Get(`$array`))
+		s.Set(`$offset`, ns.Get(`$offset`))
```

Comment:
@nevkontakte and @flimzy, I'm going to need some advice/help with this one. I can't figure out how to assign the `Value` to the new grown slice without either, modifying the `typ.nil` for the slice type, or the variable inside the `Value` isn't updated. I think that we may need to make a copy of the `nil` slice at some point, but I don't know enough about how all the JS works together to figure it out yet.

I added a test and have been using
```Bash
go1.20.14 install . && gopherjs test ./tests -v --run TestNilPrototypeNotModified
```
with generics enabled. After adding the `Object.freeze(typ.nil)` the tests return the following:
```
=== RUN   TestNilPrototypeNotModifiedByPointer
--- PASS: TestNilPrototypeNotModifiedByPointer (0.00s)
=== RUN   TestNilPrototypeNotModifiedByReflectGrow
--- FAIL: TestNilPrototypeNotModifiedByReflectGrow (0.00s)
TypeError: Cannot assign to read only property '$capacity' of object '#<typ>'
    at Object.$packages.reflect.Value.ptr.grow (/reflect/gopherjs__reflect.go:1342:4)
    at Object.$packages.reflect.Value.ptr.Grow (/reflect/value.go:2792:3)
    at TestNilPrototypeNotModifiedByReflectGrow (/github.com/Workiva/gopherjs/tests/arrays_test.go:111:3)
    ...
FAIL    github.com/gopherjs/gopherjs/tests      0.226s
```
without the freeze it will fail the capacity check.

Reply:
Actually, freezing nil values by default seems like a very good idea to catch correctness bugs. In theory, to grow a nil slice we need to allocate a new slice of the given capacity, basically the equivalent of `make([]T, 0, X)`. I'll have to poke around the code more to get more specific though.

Reply:
Okay, I think I understand what's the problem here. This fragment of the code tries to modify the slice object that's contained within `reflect.Value()`, and if that object corresponds to a nil slice, bad things happen.

What we need to do is to _replace_ the object with the new one returned by `$growSlice()` instead of copying its properties. I think all you need to do is `js.InternalObject(v.ptr).Call("$set", ns)`.

Reply:
That worked, I thought I had tried something like that but I must of done it wrong, oh well. Thank you so much for helping with that.